packages/overlay/src/integrations/sentry/data/sentryDataCache.ts

```diff
@@ -88,15 +90,15 @@ class SentryDataCache {
       });
     }
 
-    items.forEach(([itemHeader, itemData]) => {
+    for (const [itemHeader, itemData] of items) {
       if (itemHeader.type === 'event' || itemHeader.type === 'transaction') {
         (itemData as SentryEvent).platform = sdkToPlatform(sdk.name);
-        this.pushEvent(itemData as SentryEvent);
+        await this.pushEvent(itemData as SentryEvent);
```

Comment:
Not really a fan of this serial `await`s but I think we have to do it to keep the event ordering. Do you think it is safe to parallelize this with `Promise.all()`?

Reply:
I figured we insert the events into the "queue" right away and in the correct order and _then_ try to amend the frames. So I now made everything sync except for `pushEvent` which again, does the pushing first and then spins off an async request to fill in the frames.

Hoping this would still work.